api/v1/client.go

```diff
@@ -7,83 +7,96 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/dgraph-io/ristretto"
 	"github.com/go-chi/chi/v5"
-	"github.com/iancoleman/strcase"
-	oasisErrors "github.com/oasisprotocol/oasis-core/go/common/errors"
 
-	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	governance "github.com/oasisprotocol/oasis-core/go/governance/api"
 	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
-
-	"github.com/oasisprotocol/oasis-indexer/analyzer/util"
 	"github.com/oasisprotocol/oasis-indexer/api/common"
 	"github.com/oasisprotocol/oasis-indexer/log"
-	"github.com/oasisprotocol/oasis-indexer/storage"
+	storage "github.com/oasisprotocol/oasis-indexer/storage/client"
 )
 
 const (
-	tpsWindowSizeMinutes = 5
+	blockCost = 1
+	txCost    = 1
 )
 
 // storageClient is a wrapper around a storage.TargetStorage
 // with knowledge of network semantics.
 type storageClient struct {
-	db     storage.TargetStorage
+	storage *storage.StorageClient
```

Comment:
High level comment. Can you:

- remove this struct entirely
- move `blockCache` and `txCache` into `storage.StorageClient`

This way you have a very clean interface where `api` is really just the API and doesn't need to think about how or where the data is retrieved from, and stuff in `storage` handles that. Including if/when to cache data that needs greater availability/lower latency.

Reply:
Moved the cache to the storage package; though I did keep the api.storageClient for convenience for now